### PR TITLE
chore(amplify-codegen): fixes to github templates for stricter standards

### PR DESCRIPTION
**Pull request by phani-srikar**
_Description of changes:_
- Add CLI team as `CODEOWNERS` for PRs
- Changes to issue templates to align with templates in CLI repository
- Changes to PR template to enforce stricter testing standards

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/1.bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  attributes:
+    value: |
+      ## Details
+- type: dropdown
+  attributes:
+    label: Amplify Codegen Command
+    description: If applicable, what Amplify CLI commands (`amplify ...`) are surfacing the issue?
+    multiple: true
+    options:
+      - codegen
```
Comment (AaronZyLee) [inline .github/ISSUE_TEMPLATE/1.bug_report.yaml:63]: `codegen add/configure/remove` can be added to the options

File: .github/ISSUE_TEMPLATE/1.bug_report.yaml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+description: Create a report to help us improve Amplify Codegen
+title: ""
+issue_body: true
+body:
+- type: markdown
+  attributes:
+    value: Thanks for taking the time to fill out this bug report! Try to include as much information as you can. 
+- type: markdown
+  attributes:
+    value: For reference, the current version of the Amplify CLI is <img src="https://img.shields.io/npm/v/amplify-codegen.svg" />.
```
Comment (AaronZyLee) [inline .github/ISSUE_TEMPLATE/1.bug_report.yaml:11]: The image uses amplify-codegen version while the content is amplify-cli

**Review by AaronZyLee — COMMENTED** [review]
The YAML files are renamed with number added at the beginning. I am not sure this is the intended change. Otherwise LGTM 🍡

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,7 +1,34 @@
...
+
+#### Checklist
+<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
+
+- [ ] PR description included
+- [ ] `yarn test` passes
+- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/master/CONTRIBUTING.md#tests)
+- [ ] Relevant documentation is changed or added (and PR referenced)
+- [ ] Breaking changes to existing customers are released behind a feature flag or major version update
+- [ ] Changes are tested using Sample Applications for all relevant platforms (iOS/android/flutter/Javascript)
```
Comment (AaronZyLee) [inline .github/PULL_REQUEST_TEMPLATE.md:31]: Do we have some official sample apps for testing? How can third-party developers find these apps?

File: .github/ISSUE_TEMPLATE/1.bug_report.yaml [hunk 4]
```diff
@@ -0,0 +1,146 @@
+name: Bug report
+description: Create a report to help us improve Amplify Codegen
+title: ""
+issue_body: true
```
Comment (siegerts) [inline .github/ISSUE_TEMPLATE/1.bug_report.yaml:4]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/1.bug_report.yaml (hunk 2 above)
Comment (siegerts) [inline .github/ISSUE_TEMPLATE/1.bug_report.yaml:11]:
```suggestion
    value: For reference, the current version of the Amplify Codegen is <img src="https://img.shields.io/npm/v/amplify-codegen.svg" />.
```

File: .github/ISSUE_TEMPLATE/1.bug_report.yaml [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+- type: markdown
+  attributes:
+    value: |
+      > **Note**: If your bug does not result from running `amplify codegen <sub-command>`, please log it in the [Amplify CLI GitHub Issue Tracker](https://github.com/aws-amplify/amplify-cli/issues).
+- type: checkboxes
+  attributes:
+    label: |
+      Before opening, please confirm:
+    options:
+      - label: I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
```
Comment (siegerts) [inline .github/ISSUE_TEMPLATE/1.bug_report.yaml:21]:
```suggestion
      - label: I have installed the latest version of the Amplify Codegen (see above), and confirmed that the issue still persists.
```

File: .github/ISSUE_TEMPLATE/1.bug_report.yaml [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+      - codegen statements
+      - codegen types
+      - codegen models
+      - Not applicable
+  validations:
+    required: true
+- type: textarea
+  attributes:
+    label: Describe the bug
+    description: A clear and concise description of what the bug is. Markdown formatting supported.
```
Comment (siegerts) [inline .github/ISSUE_TEMPLATE/1.bug_report.yaml:73]:
```suggestion
    description: A clear and concise description of what the bug is.
```

File: .github/ISSUE_TEMPLATE/1.bug_report.yaml [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  validations:
+    required: true
+- type: textarea
+  attributes:
+    label: Expected behavior
+    description: A clear and concise description of what you expected to happen. Markdown formatting supported.
```
Comment (siegerts) [inline .github/ISSUE_TEMPLATE/1.bug_report.yaml:79]:
```suggestion
    description: A clear and concise description of what you expected to happen.
```

File: .github/ISSUE_TEMPLATE/1.bug_report.yaml [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+  validations:
+    required: true
+- type: textarea
+  attributes:
+    label: Reproduction steps
+    description: | 
+      How do you trigger this bug? Please walk us through it step by step. 
+      Screenshots can be provided in the issue body below. Markdown formatting supported.
```
Comment (siegerts) [inline .github/ISSUE_TEMPLATE/1.bug_report.yaml:87]:
```suggestion
      Screenshots can be provided in the textarea below.
```

File: .github/ISSUE_TEMPLATE/1.bug_report.yaml [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+- type: markdown
+  attributes:
+    value: |
+      ## Additional information
+- type: markdown
+  attributes:
+    value: >
+      If you have any additional information, workarounds, etc. for us, use the field below.
+      Please note, you can attach screenshots or screen recordings here, by
+      dragging and dropping files in the field below.
```
Comment (siegerts) [inline .github/ISSUE_TEMPLATE/1.bug_report.yaml:145]:
```suggestion
- type: textarea
  attributes:
    label: Additional information
    description: |
      If you have any additional information, workarounds, etc. for us, use the field below.
      Please note, you can attach screenshots or screen recordings here, by
      dragging and dropping files in the field below.
```

**Comment by siegerts** [comment]
Most of the suggestions are accounting for updates in the form functionality since the CLI base was copied.

**Comment by siegerts** [comment]
@AaronZyLee, the numbered files are for ordering when presented in the _new issue_ picker.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (phani-srikar) [reply .github/PULL_REQUEST_TEMPLATE.md:31]: We do not have one right now. Once we have them, we could direct them to the sample apps for reference. For now, they will have to create a sample app to minimally test the feature being added/modified. Updated this accordingly.

File: .github/ISSUE_TEMPLATE/1.bug_report.yaml (hunk 5 above)
Comment (phani-srikar) [reply .github/ISSUE_TEMPLATE/1.bug_report.yaml:21]: Our Codegen plugin is currently installed as part of the Amplify CLI. The codegen version that is be installed as part of CLI will be fixed major, latest minor and patch version. Currently we do not provide a command like `amplify codegen -v` for the customers to report this version. Happy to come back and change this when we add this feature.

**Review by AaronZyLee — APPROVED** [review]
LGTM 🥇

File: .github/ISSUE_TEMPLATE/1.bug_report.yaml [hunk 10]
```diff
@@ -0,0 +1,145 @@
+name: Bug report
+description: Create a report to help us improve Amplify Codegen
+title: ""
```
Comment (siegerts) [inline .github/ISSUE_TEMPLATE/1.bug_report.yaml:3]:
```suggestion

```

**Comment by siegerts** [comment]
Pinging the GH team again for this...

**Review by siegerts — APPROVED** [review]
Made one more suggested removal but LGTM after that.  Got confirmation from the GH team so the repo should be set with the new functionality.

**Pull request merged**